django/db/backends/sqlite3/operations.py

```diff
@@ -370,8 +375,27 @@ def subtract_temporals(self, internal_type, lhs, rhs):
             return 'django_time_diff(%s, %s)' % (lhs_sql, rhs_sql), params
         return 'django_timestamp_diff(%s, %s)' % (lhs_sql, rhs_sql), params
 
-    def insert_statement(self, ignore_conflicts=False):
-        return 'INSERT OR IGNORE INTO' if ignore_conflicts else super().insert_statement(ignore_conflicts)
+    def insert_statement(self, on_conflicts=ON_CONFLICTS_NONE):
+        if on_conflicts == ON_CONFLICTS_IGNORE:
+            return 'INSERT OR IGNORE INTO'
+
+        elif on_conflicts == ON_CONFLICTS_UPDATE and Database.sqlite_version_info < (3, 24, 0):
+            return 'INSERT OR REPLACE INTO'
+
+        return super().insert_statement(on_conflicts=on_conflicts)
+
+    def conflicts_suffix_sql(
+        self, opts, fields, on_conflicts=ON_CONFLICTS_NONE, update_fields=[], unique_fields=[]
+    ):
+        if on_conflicts == ON_CONFLICTS_UPDATE and Database.sqlite_version_info >= (3, 24, 0):
+            result = 'ON CONFLICT(%s) DO UPDATE SET ' % (', '.join(unique_fields))
+            result += ', '.join('%s=excluded.%s' % (field, field) for field in update_fields)
+            return result
+
+        return super().conflicts_suffix_sql(
+            opts, fields, on_conflicts=on_conflicts, update_fields=update_fields,
+            unique_fields=unique_fields
+        )
```

Comment:
```suggestion
    def on_conflict_suffix_sql(self, opts, fields, on_conflict=None, update_fields=None, unique_fields=None):
        if on_conflict == OnConflict.UPDATE and Database.sqlite_version_info >= (3, 24, 0):
            return 'ON CONFLICT(%s) DO UPDATE SET %s' % (
                ', '.join(unique_fields or ()),
                ', '.join(f'{field}=excluded.{field}' for field in update_fields or ()),
            )
        return super().on_conflict_suffix_sql(opts, fields, on_conflict, update_fields, unique_fields)
```

Same comments about feature flags and quoting apply here too.